lib/buildpacks_registry_api/client.ex

```diff
@@ -1,64 +1,309 @@
+    Jason.decode!(response, keys: :atoms)
+  end
 
-    {:ok, %HTTPoison.Response{body: body, status_code: 200}} =
-      __MODULE__.get("/search", [], params: params)
+  @doc """
+  Returns detailed information for the buildpack version.any()
 
-    body
+  ## Examples
+    iex> BuildpacksRegistryApi.Client.buildpack_version_info("heroku", "ruby", "0.1.3")
+    %{
+      addr:
+        "public.ecr.aws/heroku-buildpacks/heroku-ruby-buildpack@sha256:3eeb3773cdbd29d4fb0d578f7781fe8c525de73593480e2740b7143262e5bef5",
+      created_at: "2021-10-27T21:00:43.307Z",
+      description: "",
+      homepage: "",
+      id: "f7c0dcc3-55ab-46b9-9a89-e35ff60aa7f8",
+      licenses: nil,
+      name: "ruby",
+      namespace: "heroku",
+      stacks: ["heroku-18", "heroku-20"],
+      updated_at: "2022-03-17T21:20:26.981Z",
+      version: "0.1.3",
+      version_major: "0",
+      version_minor: "1",
+      version_patch: "3",
+      yanked: false
+    }
+  """
+  @spec buildpack_version_info(binary(), binary(), binary()) :: map()
+  def buildpack_version_info(namespace, name, version) do
+    response = get("/buildpacks/#{namespace}/#{name}/#{version}")
+    Jason.decode!(response, keys: :atoms)
   end
 
-  def buildpack_version_list(namespace, name) do
-    {:ok, %HTTPoison.Response{body: body, status_code: 200}} =
-      __MODULE__.get("/buildpacks/#{namespace}/#{name}")
+  @doc """
+  Formats the URL with query parameters.
+
+  ## Examples
+    iex> BuildpacksRegistryApi.Client.process_url("/search", %{term: "ruby"})
+    "http://localhost:9000/api/v1/search?term=ruby"
+  """
+  def process_url(path, query_params) do
+    query_params =
+      query_params
+      |> Enum.map(fn {key, value} -> "#{key}=#{value}" end)
+      |> Enum.join("&")
 
-    body
+    endpoint() <> path <> "?" <> query_params
   end
 
-  def buildpack_version_info(namespace, name, version) do
-    {:ok, %HTTPoison.Response{body: body, status_code: 200}} =
-      __MODULE__.get("/buildpacks/#{namespace}/#{name}/#{version}")
+  @doc false
+  def get(path, query_params \\ %{}) do
+    url = process_url(path, query_params)
+    url = String.to_charlist(url)
+    Logger.debug("Fetching #{url}")
 
-    body
+    {:ok, _} = Application.ensure_all_started(:inets)
+    {:ok, _} = Application.ensure_all_started(:ssl)
+
+    # https://erlef.github.io/security-wg/secure_coding_and_deployment_hardening/inets
+    cacertfile = CAStore.file_path() |> String.to_charlist()
+
+    http_options = [
+      ssl: [
+        verify: :verify_peer,
+        cacertfile: cacertfile,
+        depth: 2,
+        customize_hostname_check: [
+          match_fun: :public_key.pkix_verify_hostname_match_fun(:https)
+        ]
+      ]
+    ]
+
+    options = [body_format: :binary]
+
+    case :httpc.request(:get, {url, []}, http_options, options) do
+      {:ok, {{_, 200, _}, _headers, body}} ->
+        body
+
+      other ->
+        raise "couldn't fetch #{url}: #{inspect(other)}"
```

Comment:
Prehaps rename the function `get!` to indicate that this can raise? Tripped me up on the first pass of the code review thinking errors would get cached.

Reply:
s/get/get!/g